api/compute/v1alpha1/function_webhook.go

```diff
@@ -88,6 +88,12 @@ func (r *Function) Default() {
 		r.Spec.Namespace = DefaultNamespace
 	}
 
+	if r.Spec.SubscriptionName == "" && r.Spec.DeadLetterTopic == "" {
```

Comment:
We will need to set the subs name only if `maxMessageRetry` is set, which means the `RetryDetails` will provide to the function runtime.

Reply:
if function is created without `maxMessageRetry` or `DeadLetterTopic`, then sub is set to `tenant-namespace-name`, but when function is updated with `maxMessageRetry` and `DeadLetterTopic` provided, then the sub will be changed to `tenant/namespace/name`

look like we should set the `SubscriptionName`(like `tenant-namespace-name`) once it is not defined

Reply:
We may need to consider the backward compatibility as well, if we change the default sub name and released, when users upgrade function-mesh, their existing function pods may change the sub name and cause duplicate consumption

looks like the better way is to set the deadLetterTopic name